example/azure-test-garden/README.md

```diff
@@ -0,0 +1,530 @@
+     * [Troubleshooting Shoot Creation Issues](#troubleshooting-shoot-creation-issues)
+5. [Access Shoot cluster](#access-shoot-cluster)
+6. [Delete Shoot cluster](#delete-shoot-cluster)
+
+# Prerequisites
+
+Summary of prerequisites:
+- An Azure AKS cluster with:
+  - default ClusterRoles defined (not defined by default since RBAC is not enabled),
+  - helm(tiller) deployed with cluster-admin ClusterRole,
+  - stable/nginx-ingress helm chart deployed,
+- An Azure service principle to provide Azure credentials to gardener,
+- A Route53 Zone and AWS account credentials with permissions on that Route53 Zone.
+- gardenctl command configured for the cluster
+
+**Note**: We use a Route53 Zone even if we are deploying on Azure, this is due
+to lack of Azure DNS implementation in Gardener, it only supports
+aws-route53 or google-clouddns for now (TODO: Needs to be verified).
+
+## Deploy AKS cluster
+
+Here you can find a summary for creating an AKS cluster, if you already
+have one, skip this. You can **skip the Azure HTTP addons** when
+installing the AKS cluster, we'll install stable/nginx-ingress helm chart as
+you can see below.
+
+```
+az group create --name garden-1 --location eastus
+az aks create --resource-group garden-1 --name garden-1 --node-count 3 --generate-ssh-keys -s Standard_DS4_v2
+az aks get-credentials --resource-group garden-1 --name garden-1
+```
+
+### Install default ClusterRoles
+
+On Azure, RBAC support is planned for GA and not enabled yet, so
+ClusterRole definisions are not in place and this causes Gardener helm
+chart to fail. Below instead of removing RBAC support from Gardener
+installation we are faking the RBAC support by manually creating ClusterRoles
+on our AKS cluster.
+
+**Note**: We are using v1.9.6 in our AKS cluster, mind the version in
+the url below and please use the rules for the right version.
+
+```
+# manually create ClusterRoles in AKS, required until RBAC support is added to AKS
+kubectl apply -f https://raw.githubusercontent.com/kubernetes/kubernetes/v1.9.6/plugin/pkg/auth/authorizer/rbac/bootstrappolicy/testdata/cluster-roles.yaml --validate=false
+```
+
+### Install helm/tiller to AKS
+
+Helm is also need to be run with a ClusterRole (cluster-admin) with
+enough permissions even if RBAC is not enabled, this is an enforcement
+needs to be addressed on Helm side not on AKS/Kubernetes. So lets
+create a tiller ServiceAccount with cluster-admin ClusterRole and
+deploy Helm with the ServiceAccount.
+
+```
+# Reference: https://github.com/kubernetes/helm/blob/master/docs/rbac.md
+kubectl apply -f example/azure-test-garden/tiller-rbac-config.yaml
+helm init --service-account tiller
+```
+
+### Deploy stable/nginx-ingress chart to AKS
+
+Instead of using the ingress provided by Azure we deploy the default
+nginx-ingress using the helm chart. We cn deploy the chart
+
+TODO: why is this needed, and is this related to
```

Comment:
At the moment all `Ingress` resources created by the Gardener are using the `nginx` annotations. This is the simplest solution at the moment, because we can use it across cloud providers, but in the future we plan to make it more customizable.